### PR TITLE
Mailbox 270: old behaviour

### PR DESCRIPTION
**Pull request by quynhn**
For now, just saw the delete mailbox should be side effect.
- All the mailbox's annotation should be deleted when mailbox is deleted

### CODE REVIEW

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            } catch (MailboxException e) {
+                logger.error("Unable to look up AnnotationMapper", e);
+            }
+        }
+    }
+
+    private void deleteRelatedAnnotations(MailboxId mailboxId, AnnotationMapper annotationMapper) {
+        List<MailboxAnnotation> annotations = annotationMapper.getAllAnnotations(mailboxId);
+        for (MailboxAnnotation annotation : annotations) {
+            annotationMapper.deleteAnnotation(mailboxId, annotation.getKey());
```
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: what about trying to delete all annotations even if one deletion failed ?

Comment (quynhn) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: For now, it does not catch any exception from Mapper layer so all the annotation should be still there if one deletion failed. 
I tried to add more testing but from mapper layer, no way to throw Exception. Should we accept RuntimeException and keep duty data in that case?

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.mail.AnnotationMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class MailboxAnnotationListener implements MailboxListener {
+    private final Logger logger = LoggerFactory.getLogger(MailboxAnnotationListener.class);
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: static

Comment (quynhn) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: Done

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java (hunk 1 above)
Comment (mbaechler) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: I don't understand why you can't throw at mapper layer ?
Anyway, switch to runtime exception is ok if needed.

Comment (quynhn) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxAnnotationListener.java]: Done

**Comment by mbaechler** [comment]
👍

**Comment by aduprat** [comment]
:+1:

**Comment by quynhn** [comment]
test this please

**Pull request merged**